mantidproject/mantid
# MergeRuns to handle merging sample logs

This PR adds options to MergeRuns to deal with the sample logs. Behaviour can be to merge as a time series (numeric values), merge as a list, warn if values are different and fail if values are different.

How to deal with each value is defined in either the instrument parameters file or as arguments to MergeRuns.

**To test:**

Functional:

Either pick your favourite NeXus files or grab some ILL IN6 files from here -  https://www.ill.eu/fileadmin/users_files/img/instruments_and_support/support_facilities/computing_for_science/Computing_for_Science/Data_analysis/IN6_Script_Data.zip

Pick some parameters and try adding them to the instrument parameters file (see updated MergeRuns-v1.rst for example XML for IN6 files).

Do the same with setting the options via the MergeRuns algorithm.

Check documentation updates make sense.

Code:

Check unit test pass. Any obvious unit tests missing for the new behaviour?  Would be useful to review the code too, but there is quite a lot there.



Fixes #16928.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
